src/internal/schema.ts

```diff
@@ -65,7 +65,7 @@ export function _buildSchema<T, U extends AnyParamConstructor<T>>(
     sch = new Schema(schemas.get(name), schemaOptions);
   } else {
     sch = sch.clone();
-    sch.add(schemas.get(name));
+    sch.add(schemas.get(name)!);
```

Comment:
why is an extra `!` here after everything?

Reply:
Because a call to a `map.get()` can return undefined if the key doesn't exist, but this line above:
https://github.com/typegoose/typegoose/pull/254/files/b0bb10c3b10b63e7fba3cad4b45248251984d74b#diff-47432c0fca2331f971956d7674048e26L61

Ensures that the key would be there, so `!` is an override to inform the compiler that the result cannot be undefined.

Reply:
thanks